File: openapi/code_samples/csharp/api_company_find.cs
using Microsoft.AspNetCore.Mvc;
using Swashbuckle.AspNetCore.Annotations;

[ApiController]
[Route("api/[controller]")]
public class CompanyController : ControllerBase
{
    [HttpGet]
    [Route("find")]
    [SwaggerOperation(Summary = "Retrieves companies based on CUI or name in a pageable manner.")]
    [ProducesResponseType(typeof(PageCompanyDetailsResponse), 200)]
    [ProducesResponseType(typeof(Fault), 400)]
    public IActionResult Find(
        [FromQuery(Name = "pageNo")] int pageNo = 0,
        [FromQuery(Name = "pageSize")] int pageSize = 10,
        [FromQuery(Name = "cuiOrName")] string cuiOrName = "",
        [FromQuery(Name = "sortBy")] string sortBy = "name",
        [FromQuery(Name = "currency")] string currency = "RON")
    {
        // Your implementation here
        return Ok();
    }
}

File: openapi/code_samples/python/api_company_find.py
from flask import Flask, request, jsonify

app = Flask(__name__)

@app.route('/api/company/find', methods=['GET'])
def find():
    pageNo = int(request.args.get('pageNo', 0))
    pageSize = int(request.args.get('pageSize', 10))
    cuiOrName = request.args.get('cuiOrName', '')
    sortBy = request.args.get('sortBy', 'name')
    currency = request.args.get('currency', 'RON')

    # Your implementation here

    return jsonify({}), 200

if __name__ == '__main__':
    app.run()
